### PR TITLE
cgu: add wait until backup starts function

### PR DESCRIPTION
**Pull request by klaskosk**

### CODE REVIEW

File: pkg/cgu/cgu.go [hunk 1]
```diff
@@ -344,3 +344,36 @@ func (builder *CguBuilder) WaitUntilComplete(timeout time.Duration) (*CguBuilder
...
+
+		return builder, fmt.Errorf(builder.errorMsg)
+	}
+
+	var err error
+	err = wait.PollUntilContextTimeout(context.TODO(), 3*time.Second, timeout, true, func(context.Context) (bool, error) {
+		builder.Object, err = builder.apiClient.RanV1alpha1().ClusterGroupUpgrades(builder.Definition.Namespace).
+			Get(context.TODO(), builder.Definition.Name, metav1.GetOptions{})
+		if err != nil {
+			return false, nil
```
Comment (kononovn) [inline pkg/cgu/cgu.go:368]: Please add glog.V if error occurs

File: pkg/cgu/cgu.go [hunk 2]
```diff
@@ -344,3 +344,36 @@ func (builder *CguBuilder) WaitUntilComplete(timeout time.Duration) (*CguBuilder
...
+		}
+
+		return builder.Object.Status.Backup != nil, nil
+	})
+
+	if err == nil {
+		return builder, nil
+	}
+
```
Comment (kononovn) [inline pkg/cgu/cgu.go:377]: Please add glog.V here as well

**Comment by klaskosk** [comment]
@kononovn Added the logs and a test for the function. I modeled it after `TestWaitUntilCondition` in `deployment/deployment_test.go` but I'm not sure if it needs more test cases

File: pkg/cgu/cgu.go (hunk 1 above)
Comment (kononovn) [reply pkg/cgu/cgu.go:370]: Please print error itself in the log as well. Add "due to: %w"

File: pkg/cgu/cgu.go (hunk 2 above)
Comment (kononovn) [reply pkg/cgu/cgu.go:379]: Please print error itself in the log as well. Add "due to: %w"

**Review by kononovn — CHANGES_REQUESTED** [review]
Please fix few comments above.

**Comment by klaskosk** [comment]
@kononovn Thanks, I've updated the logs to include the error

**Review by kononovn — APPROVED** [review]
/lgtm

**Review by achuzhoy — APPROVED** [review]

**Pull request merged**